x-pack/plugins/apm/server/lib/helpers/get_apm_event_client.ts

```diff
@@ -32,7 +32,7 @@ export async function getApmEventClient({
     ]);
 
     return new APMEventClient({
-      esClient: coreContext.elasticsearch.client.asInternalUser,
+      esClient: coreContext.elasticsearch.client.asCurrentUser,
```

Comment:
As mentioned on Slack, we should have an api test that verifies that only users with access to the apm indices can view APM data.

Reply:
The internal user has read access to the APM indices

Reply:
We can add a test however that runs with a user that has no access to APM data?

Reply:
> The internal user has read access to the APM indices

Exactly. So if we add a test with a user that _does not_ have access to APM indices, they should not be able to retrieve APM data.

Reply:
https://github.com/elastic/kibana/issues/145516